### PR TITLE
feat: Notify customers if newer submitter is available.

### PR DESCRIPTION
**Pull request by karthikbekalp**
### What was the problem/requirement? (What/Why)

Customers did not know whether they were on the latest submitter or not. This means they would not have the latest bug fixes or features to work with our product. 

### What was the solution? (How)

This adds a dialog which checks our download manifests and if there's a new update available notifies customer  that. 

Depends on: https://github.com/aws-deadline/deadline-cloud/pull/1070

image

Customers can deactivate the feature in the settings. 

image

### What is the impact of this change?

Customers hopefully are always on the latest updated submitter. 

### How was this change tested?

| # | Test Case | Steps | Expected Result | Pass? |
|---|-----------|-------|-----------------|-------|
| | **Config GUI — setting visible** | | | |
| 1 | Setting in config GUI | Open Blender → Deadline Cloud → Settings | "Show submitter update notifications" checkbox is visible | YES |
| | **Config disabled — no dialog** | | | |
| 2 | Notification disabled via GUI | Uncheck "Show submitter update notifications" → Apply → close & reopen submitter | No network call, no dialog, submitter opens normally | YES |
| | **Config enabled, no internet — graceful failure** | | | |
| 3 | Re-enable notification via GUI | Check "Show submitter update notifications" → Apply | Setting updated | YES |
| 4 | No internet / firewall | Disconnect network → open submitter | Submitter opens normally, no crash, no visible error | YES |
| | **Config enabled, internet available, no update** | | | |
| 5 | Current == latest | Restore network, ensure installed version matches manifest | No dialog, submitter opens normally | YES |
| 6 | Current > latest (dev build) | Install a version newer than manifest | No dialog, submitter opens normally | YES |
| | **Config enabled, internet available, update exists** | | | |
| 7 | Current < latest | Install an older version than manifest | Update dialog appears | YES |
| 8 | Dialog content | Inspect the dialog | Shows "Blender", correct current & latest versions | YES |
| 9 | Dialog layout | Resize / check | Min width 400px, text wraps properly | YES |
| 10 | Dialog timing | Observe launch order | Dialog appears before Blender submitter window | YES |
| | **Dialog interactions — Dismiss** | | | |
| 11 | Click "Dismiss" | Click Dismiss button | Dialog closes, submitter opens normally | YES |
| 12 | Close via X | Click X button on dialog | Dialog closes, submitter opens normally | YES |
| | **Dialog interactions — Download** | | | |
| 13 | Click "View release notes" | Click the link | Blender GitHub releases page opens in browser | YES |
| 14 | Click "Download" | Click Download button | Correct installer URL opens in browser | YES |
| 15 | Restart reminder | After Download click | "Application Restart Required" info box appears | YES |
| | **Edge cases** | | | |
| 16 | Non-standard `_version.py` tuple | Modify version tuple | No crash | YES |

#### Please run the integration tests and paste the results below

```
test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_start_and_close_blender 
[gw0] [ 12%] PASSED test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_start_and_close_blender 
test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_cancel_closes_blender 
[gw0] [ 25%] PASSED test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_cancel_closes_blender 
test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
[gw0] [ 37%] PASSED test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
[gw0] [ 50%] PASSED test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter 
[gw0] [ 62%] PASSED test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter 
test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
[gw0] [ 75%] PASSED test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop 
[gw0] [ 87%] PASSED test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop 
test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command 
[gw0] [100%] PASSED test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command 

================================================================= slowest 5 durations ==================================================================
27.53s call     test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor
13.24s call     test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor
9.04s call     test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command
7.82s call     test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter
7.77s call     test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop
============================================================= 8 passed in 84.37s (0:01:24) =============================================================
```

#### If `installer/` was modified or a file was added/removed from `src/`, then update the installer tests and post the test results below

No

### Was this change documented?

Yes

### Is this a breaking change?

No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Comment by jericht** [comment]
@karthikbekalp what issues did you have running the integ tests?

**Comment by jericht** [comment]
> Depends on: https://github.com/aws-deadline/deadline-cloud/pull/1070

Should be released now, let me check

**Comment by karthikbekalp** [comment]
> @karthikbekalp what issues did you have running the integ tests?

I've not run the integration tests right now. (Not sure if its required because I don't think integration tests would cover this no?) 

> Should be released now, let me check

Yup, 0.55.1 was released last week.

**Comment by jericht** [comment]
> I've not run the integration tests right now. (Not sure if its required because I don't think integration tests would cover this no?)

Let's run them. The changes in the add on's `__init__.py` file can impact loading up the submitter in Blender.

**Review by jericht — APPROVED** [review]
Let's make sure integ tests are passing before we merge just in case 🙏

**Comment by karthikbekalp** [comment]
Hi @jericht , 

I tried running the integration tests but some of them failed. 


```
test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_start_and_close_blender 
[gw0] [ 12%] PASSED test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_start_and_close_blender 
test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_cancel_closes_blender 
[gw0] [ 25%] PASSED test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_cancel_closes_blender 
test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
[gw0] [ 37%] PASSED test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
[gw0] [ 50%] PASSED test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter 
[gw0] [ 62%] FAILED test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter 
test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
[gw0] [ 75%] FAILED test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop 
[gw0] [ 87%] PASSED test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop 
test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command 
[gw0] [100%] PASSED test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command 
```
The tests that failed are related to add on failures:

```
Add-on not loaded: "deadline_cloud_blender_submitter", cause: No module named 'deadline_cloud_blender_submitter'
```

Any ideas how I can unblock myself?

**Comment by karthikbekalp** [comment]
Never mind, I installed the add-on and it looks like it worked. 

```

test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_start_and_close_blender 
[gw0] [ 12%] PASSED test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_start_and_close_blender 
test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_cancel_closes_blender 
[gw0] [ 25%] PASSED test/integ/test_adaptor_lifecycle.py::TestAdaptorLifecycle::test_adaptor_cancel_closes_blender 
test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
[gw0] [ 37%] PASSED test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
[gw0] [ 50%] PASSED test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter 
[gw0] [ 62%] PASSED test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter 
test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
[gw0] [ 75%] PASSED test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop 
[gw0] [ 87%] PASSED test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop 
test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command 
[gw0] [100%] PASSED test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command 

================================================================= slowest 5 durations ==================================================================
27.53s call     test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_adaptor
13.24s call     test/integ/test_blender_adaptors.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor
9.04s call     test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_run_command
7.82s call     test/integ/test_blender_submitters.py::TestSubmitters::test_minimal_scene_submitter
7.77s call     test/integ/test_daemon_mode.py::TestDaemonMode::test_daemon_start_and_stop
============================================================= 8 passed in 84.37s (0:01:24) =============================================================
```

**Review by godobyte — APPROVED** [review]

File: src/deadline/blender_submitter/addons/deadline_cloud_blender_submitter/update_utils.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    Returns:
+        True if the user clicked Download (caller should skip opening the submitter),
+        False otherwise.
+    """
+    if _session_state.update_dismissed:
+        return False
+
+    update_result = _check_for_update()
+    if update_result.status == UpdateCheckStatus.SUCCESS and update_result.update_available:
```
Comment (godobyte) [inline src/deadline/blender_submitter/addons/deadline_cloud_blender_submitter/update_utils.py:54]: Curious - what would be the case then `update_result.status` is not SUCCESS? Do we need to add a note or display somehow?

Comment (karthikbekalp) [reply src/deadline/blender_submitter/addons/deadline_cloud_blender_submitter/update_utils.py:54]: That's a good point. But I think it should be in the helper functions instead. 
I'll add some debug logs so that if customers have debug logging enabled they will know about the failures. https://github.com/aws-deadline/deadline-cloud/pull/1112/changes
